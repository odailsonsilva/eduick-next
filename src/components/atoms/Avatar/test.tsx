import { screen } from '@testing-library/react'
import { renderWithTheme } from '../../../utils/tests/helpers'

import { Avatar } from '.'

describe('<Avatar />', () => {
  it('shold render the small size default', () => {
    renderWithTheme(<Avatar />)
    expect(screen.getByRole('img', { name: /profile/i })).toBeInTheDocument()
  }),
    it('shold render the large size', () => {
      const { container } = renderWithTheme(<Avatar size="large" />)
      const image = screen.getByRole('img', { name: /profile/i })

      expect(image).toBeInTheDocument()
      expect(image.parentElement?.parentElement).toHaveStyle({
        height: '5rem',
        width: '5rem'
      })
      expect(container.firstChild).toMatchSnapshot()
    })
})
