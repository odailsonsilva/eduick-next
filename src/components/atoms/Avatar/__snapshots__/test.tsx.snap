// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> shold render the large size 1`] = `
.c0 {
  height: 5rem;
  width: 5rem;
  position: relative;
  background: #f6f7f8;
  background-image: linear-gradient( to right,#f6f7f8 0%,#edeef1 20%,#f6f7f8 40%,#f6f7f8 100% );
  background-size: 80rem 14rem;
  -webkit-animation: placeholderShimmer 1s linear infinite forwards;
  animation: placeholderShimmer 1s linear infinite forwards;
  border-radius: 50%;
}

.c0::after {
  content: '';
  width: 0.96rem;
  height: 0.96rem;
  background: #FFD74F;
  border-radius: 50%;
  position: absolute;
  top: -1px;
  right: 2px;
}

.c0 img {
  border-radius: 50%;
}

<div
  class="c0"
>
  <div
    style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
  >
    <noscript />
    <img
      alt="Profile"
      data-nimg="true"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
    />
  </div>
</div>
`;
